Polyhedron/demo/Polyhedron/Plugins/AABB_tree/Cut_plugin.cpp

```diff
@@ -1204,14 +1204,33 @@ void Polyhedron_demo_cut_plugin::createCutPlane() {
     CGAL::Three::Scene_item* item = scene->item(i);
     Scene_polyhedron_item* poly_item = qobject_cast<Scene_polyhedron_item*>(item);
     if(!poly_item) continue;
-
+    if(!poly_item->polyhedron()->is_pure_triangle())
+    {
+      messages->warning(QString("%1 ignored (not pure triangle)").arg(poly_item->name()));
+      continue;
+    }
     if(facet_trees.find(poly_item) == facet_trees.end()) {
       facet_trees[poly_item] = new Facet_tree();
       PPMAP pmap;
-      facet_trees[poly_item]->insert(faces(*(poly_item->polyhedron())).first,
+      //filter facets to ignore degenerated ones
+      for(Polyhedron::Facet_iterator fit = poly_item->polyhedron()->facets_begin(), end = poly_item->polyhedron()->facets_end(); fit!=end; ++fit)
+      {
+        Polyhedron::Point a(fit->halfedge()->vertex()->point()),
+            b(fit->halfedge()->next()->vertex()->point()),
+            c(fit->halfedge()->prev()->vertex()->point());
+
+        Polyhedron::Traits::Collinear_3  collinear = poly_item->polyhedron()->traits().collinear_3_object();
+        bool test = collinear(a,b,c);
+        if(!test)
```

Comment:
Well, Maxime has learned so far to write CGAL generic code, and in a generic code, all access to the geometry should be through geomtraits functors.

Here the code is wrote only for a given type, a `CGAL::Polyhedron_3` with a standard CGAL kernel, and that is the reason why one can use the global function `CGAL::collinear`.